Design/Mantid4ReleasePlanning.md

```diff
@@ -0,0 +1,65 @@
+Release Planning for Mantid version 4.0
+=======================================
+
+Motivation
+----------
+
+We have a number of desired changes that will break backward compatibility for users scripts, and potentially ways of working with 
+Mantid and Mantidplot.  Following the rules of Semantic versioning (http://semver.org/) this means we need to increase the major version number. While we can make these kind of changes in a planned way with carefull communication, we cannot follow that up
+with other breaking changes soon afterwards.
+
+In order to decide on the changes to make it into this major release, and to align them to deliver together we will document the planned changes
+to include here.
+
+Timing of Mantid 4.0
+--------------------
+
+The timing for Mantid 4.0 is not determined by the current version numbering, that is to say there is no reason that 4.0 should follow 3.9, we could perfectly well have version 3.10 and so on.
+The determinating step for Mantid 4.0 is the list of breaking changes below.  If we decide all of them are to remain in this release, then the latest to deliver will determine the timng of Mantid 4.0.
+
+User Expectations of Mantid 4.0
+-------------------------------
+
+We have not raised the concept of Mantid 4.0 with our users, so we have not made any commitment about what it may contain.  However we should be aware that while it is  not defined in Semantic versioning (http://semver.org/), the users will expect some significant improvements in order to justify the version change, and to act as a carrot to entice them to upgrade and overcome the discomfort of the breaking changes we are introducing.  Some of the breaking changes themselves will bring significant advantages, but the benefits of others are less visible.  It is worth considering other significant improvements that can be aligned with Mantid 4.0 to encourage enthusiastic adoption of these changes.
+
+List of breaking changes 
+------------------------
+
+1. Change from qwt5 graphs to ? [3 people 1-3m]
+1. Move to Python 3 on all platforms [1 person 2m]
```

Comment:
Building RHEL 7 RPMs for all Mantid dependencies using python 3 is a significant undertaking. @stuartcampbell might have a better idea of the time required.